### PR TITLE
Code refactor in Catalog ViewModel Breadcrumbs

### PR DESCRIPTION
**Pull request by srsathish92**
### Description (*)
This PR helps to refactor the code of ViewModel Breadcrumbs in Catalog Module & removed unused JSON Class, utilised JsonHexTag class for replacing PHP json_encode with JsonHexTag option, Also updated the test case with JsonHexTag class.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php [hunk 1]
```diff
@@ -39,20 +44,21 @@ class Breadcrumbs extends DataObject implements ArgumentInterface
...
-     * @param Json|null $json
+     * @param JsonHexTag $jsonSerializer
      * @param Escaper|null $escaper
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function __construct(
         Data $catalogData,
         ScopeConfigInterface $scopeConfig,
-        Json $json = null,
+        JsonHexTag $jsonSerializer,
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php]: Original class was kept for backward compatibility reasons, please do not remove it.
Could you introduce new optional parameter for JsonHexTag object?

File: app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php [hunk 2]
```diff
@@ -101,15 +107,14 @@ public function getProductName(): string
...
+        return $this->jsonSerializer->serialize(
             [
                 'breadcrumbs' => [
                     'categoryUrlSuffix' => $this->escaper->escapeHtml($this->getCategoryUrlSuffix()),
                     'useCategoryPathInUrl' => (int)$this->isCategoryUsedInProductUrl(),
                     'product' => $this->escaper->escapeHtml($this->getProductName())
                 ]
-            ],
-            JSON_HEX_TAG
+                ]
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php]: Looks like formatting is little bit broken. Please fix it
```suggestion
            ]
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php (hunk 2 above)
Comment (srsathish92) [reply app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php]: @ihor-sviziev updated

File: app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php (hunk 1 above)
Comment (srsathish92) [reply app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php]: @ihor-sviziev Thanks for the suggestion, As suggest I have updated.

Comment (ihor-sviziev) [reply app/code/Magento/Catalog/ViewModel/Product/Breadcrumbs.php]: Please make new argument as optional, move to last position and get it from ObjectManager as fallback

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6441](https://jira.corp.magento.com/browse/ENGCOM-6441) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@srsathish92 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**